### ISSUE TITLE
Questions about mask_ratio, mask_overlap, pretrained and augmentation process  

### ISSUE THREAD

**Issue by DP1701** [ultralytics/ultralytics#1458]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello all,

Thanks for the release of the new YOLO version. 

I currently have a few questions that the documentation can't answer. 

Segmentation:

What is the function of the parameter "mask_ratio"? What does the 4 mean and what changes if I enter a 1?

What is the function of the parameter "overlap_mask"? Is there an overlap of the masks when using copy_paste (augmentation)?

What does the parameter "pretrained" stand for? Isn't a pre-trained model used during training, for example if I enter model=yolov8n-seg.pt?

Augmentation:

Is the current albumentations implementation compatible with training mode "segment"?

**Comment by AyushExel** [comment]
Please use the yolo command for training.
```
yolo segment train device="1,2,3,4"
```
https://docs.ultralytics.com/tasks/segment/

**Comment by DP1701** [comment]
Thanks for the answer. 

I think that albumentations is not usable in connection with the task "Segment", right? 

Another thing I noticed is that the autobatch function (batch=-1) does not work with Multi GPU training.
My current workaround is:
1. test it on a single GPU
2. note the result of the autobatch function
3. multiply it with the number of GPUs
4. run a multi GPU training with the calculated batch size

**Comment by glenn-jocher** [comment]
@DP1701 yes that sounds like a good plan. That might be the simplest workflow for Multi-GPU AutoBatch :)

**Comment by glenn-jocher** [comment]
@DP1701 segment training should work correctly with Albumentations. See https://albumentations.ai/docs/getting_started/mask_augmentation/

**Comment by DP1701** [comment]
@glenn-jocher It's true that allbuminations offers this but has this been taken into account in YOLOv8?

When I see the current [implementation](https://github.com/ultralytics/ultralytics/blob/ef03e6732aecc91134275a3f5a62f2e0f82ab4bf/ultralytics/yolo/data/augment.py#L568) of allbuminations in YOLOv8, only the bounding boxes ([format=yolo](https://albumentations.ai/docs/getting_started/bounding_boxes_augmentation/#yolo)) are considered, or am I mistaken here?

On the website it is also written:
<img width="1051" alt="Bildschirmfoto 2023-03-25 um 23 42 06" src="https://user-images.githubusercontent.com/68444979/227745767-cfd50f8b-edc5-4505-9466-34b1695849ce.png">

If only pixel-level transforms are used, this should probably work. It might become problematic if spatial-level transforms are used. 

I have one more question. Currently YOLOv8 should be installed via pip. With YOLOv5 the pipline could be extended directly in the source code. How does it look with YOLOv8? Is something provided for this? Or do I have to download the repo, extend the pipline in source code and create a python package later?

**Comment by glenn-jocher** [comment]
@DP1701 yes it's possible additional steps are required to use masks with Albumentations in YOLOv8. 

You can also git clone YOLOv8 if you'd like. See https://docs.ultralytics.com/quickstart for details.

**Comment by vittorio-prodomo** [comment]
> mask_ratio: It is used to set the downsampling ratio when training the segmentation mask. 4 means the masks/images will be downsampled by a factor of 4. We're using it by default as it is much faster to train and it doesn't lose much accuracy. Setting it to 1 will train in the original resolution.
> 
> mask_overlap: This trains the segmentation models by overlapping all individual masks as one mask. This also improves speed by a lot (more than doubles the speed sometimes!). Disabling it might give you a slight improvement in map.
> 
> Pretrained: Yes this is a new addition for segmentation and detection. Your usage is correct. if you pass model=model.pt, it will be used as a pre-trained model. now you can use pretrained=model.pt to transfer pre-trained weights to a model. See this for usage [#1406 (comment)](https://github.com/ultralytics/ultralytics/issues/1406#issuecomment-1471968676)

Sorry, can I ask for more elaborate explanations of these? I am also interested in understanding. 

mask_ratio: from your explanation here, it seems it also downsamples images, but is that the case? I thought images were only downsampled to imgsz, and this ratio only downsampled masks with the logic of "usually masks don't need that pixel-level accuracy, we can work well also with masks that are more approximated". So can we have a better explanation of this parameter?

mask_overlap: neither this nor the official docs explanation really makes it clear what this does, in my opinion. For reference, the official docs say: "Determines whether object masks should be merged into a single mask for training, or kept separate for each object. In case of overlap, the smaller mask is overlaid on top of the larger mask during merge.". Can we please get a better explanation? Lots of ambiguities here. What did you mean with "overlapping all individual masks as one mask"? That we lose the "per-instance" information? Is this done per-class, or class-agnostic? I mean, please can someone explain this better?

Pretrained: so, up until now I believed I could use pre-trained weights simply by passing the .pt model as the "model" parameter. In that case, I thought a simple value of True is enough for the "pretrained" argument. So when is it actually useful to provide "pretrained" as a string (path to a model) instead of a boolean? Maybe when we are passing a model as yaml? Can someone confirm this for me?

**Comment by glenn-jocher** [comment]
@vittorio-prodomo great questions! Let me clarify each parameter:

**mask_ratio**: This only downsamples the segmentation masks, not the images. Images are still resized to `imgsz` as usual. The masks are processed at a lower resolution (1/4 by default) for computational efficiency during training while maintaining good accuracy.

**mask_overlap**: When `True`, all individual object masks in an image are merged into a single combined mask for training. This loses per-instance information but significantly speeds up training. When `False`, each object mask is kept separate, preserving instance-level details but requiring more computation.

**pretrained**: Use the string path when building from YAML config files. For example: `model="yolo11n-seg.yaml"` with `pretrained="yolo11n-seg.pt"` transfers pretrained weights to the new architecture. When using `model="yolo11n-seg.pt"` directly, `pretrained=True` is sufficient since the model file already contains the weights.

You can see the [train settings documentation](https://docs.ultralytics.com/modes/train/) for more details on these parameters.
